### PR TITLE
Modify assignment UI

### LINKED ISSUE #229: Leave feedback/replies inline on question screen
A teaching assistant giving feedback to multiple students in the application had a lot of lag time. He had to navigate to the reply screen, reply and submit, then wait for student answers to load again on the question screen, select another answer to reply to, and continue the cycle.

It may be preferable to allow all users to reply inline on the answer screen, instead of forcing a jump to another screen for the reply and then returning to the answer screen after submitting.

### PR DESCRIPTION
**Pull request by andrew-gardener**
Allow users to:
- Edit answers inline (modal)
- Create new answer comments inline (modal)
- Edit answer comments inline (modal)
- Create assignment (help) comments inline (modal)
- Edit assignment (help) comments inline (modal)

This effects the 'Answers', 'Help', and 'Your Work + Feedback' tabs

Note: Creating an answer still open into a new page

Also added a new directive for modal cancel button

Fixes #229

### CODE REVIEW

**Comment by lenglund** [comment]
Does this mean the modals don't apply to the student work & feedback tab?

**Comment by andrew-gardener** [comment]
Currently yes. Do you want modals added for all the tabs on the assignment page?

**Comment by lenglund** [comment]
Might be nice to have the edit answer for that tab, but not a requirement.

**Comment by andrew-gardener** [comment]
Added modal for the edit answer on the 'Your Work + Feedback' as well as create/edit assignment comments on the 'help' tab

File: acj/static/modules/answer/answer-create-modal-partial.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<modal-cancel-button modal-instance="modalInstance" ng-show="modalInstance !== undefined"></modal-cancel-button>
```
Comment (xcompass) [inline acj/static/modules/answer/answer-create-modal-partial.html]: Looks like answer create and edit partial have only one difference. Could you use one and pass the text from the controller to reduce the duplicate code?

File: acj/static/modules/comment/comment-assignment-create-modal-partial.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<modal-cancel-button modal-instance="modalInstance" ng-show="modalInstance !== undefined"></modal-cancel-button>
```
Comment (xcompass) [inline acj/static/modules/comment/comment-assignment-create-modal-partial.html]: Save as above.

File: acj/static/modules/answer/answer-create-modal-partial.html (hunk 1 above)
Comment (andrew-gardener) [inline acj/static/modules/answer/answer-create-modal-partial.html]: amended

File: acj/static/modules/comment/comment-assignment-create-modal-partial.html (hunk 2 above)
Comment (andrew-gardener) [inline acj/static/modules/comment/comment-assignment-create-modal-partial.html]: amended

**Pull request merged**